viadot/tasks/sap_rfc.py

```diff
@@ -116,8 +132,22 @@ def run(
         Returns:
             pd.DataFrame: DataFrame with SAP data.
         """
-        if query is None:
-            raise ValueError("Please provide the query.")
+
+        if isinstance(credentials, dict):
+            credentials_keys = list(credentials.keys())
+            required_credentials_params = ["sysnr", "user", "passwd", "ashost"]
+            for key in required_credentials_params:
+                if key not in credentials_keys:
+                    self.logger.warning(
+                        f"Required key '{key}' not found in your 'sap_credentials' dictionary!"
+                    )
+                    credentials = None
+
+        if credentials is None:
+            credentials_str = AzureKeyVaultSecret(
+                secret=saprfc_credentials_key,
+            ).run()
+            credentials = json.loads(credentials_str).get(env)
```

Comment:
Please add some test to this part and the rest new 'features'. Validate the output without with different setup of the local environment to handle the possible issues with Azure services or missing keys in KV.